mapcomposer/app/static/externals/gxp/src/script/plugins/WMSSource.js

```diff
@@ -510,21 +512,55 @@ gxp.plugins.WMSSource = Ext.extend(gxp.plugins.LayerSource, {
         return compatibleProjection;
     },
     
+    /** private: method[switchWmsVersion]
+     *  returns the WMS version alternative to the one passed as parameter
+     */
+    switchWmsVersion: function(wmsVersion) {
+    	var WMS_VERSIONS = {
+    		"1.1.1": "1.3.0",
+    		"1.3.0": "1.1.1"
+    	};
+    	
+    	return WMS_VERSIONS[wmsVersion];
+    },
+    
     /** private: method[initDescribeLayerStore]
      *  creates a WMSDescribeLayer store for layer descriptions of all layers
      *  created from this source.
      */
     initDescribeLayerStore: function() {
-        var req = this.store.reader.raw.capability.request.describelayer;
-        if (req) {
-            this.describeLayerStore = new GeoExt.data.WMSDescribeLayerStore({
-                url: req.href,
-                baseParams: {
-                    VERSION: this.store.reader.raw.version,
-                    REQUEST: "DescribeLayer"
-                }
-            });
-        }
+    	var DEFAULT_WMS_VERSION = "1.1.1";
+    	var url;
+    	var version;
+    	var baseParams = {
+    			SERVICE: "WMS",
+    			REQUEST: "DescribeLayer"
+    		};
+    	
+    	if (this.useCapabilities === true) {
+    		var describeLayerRequest = this.store.reader.raw.capability.request.describelayer;
+    		var wmsVersion = this.store.reader.raw.version;
+    		
+    		if (describeLayerRequest) {
+    			url = describeLayerRequest.href;
+    			version = wmsVersion; 
+    		} else {
+    			url = this.url;
+    			version = this.switchWmsVersion(wmsVersion);
```

Comment:
Well, I did it on purpose because I thought the code was more general that
way... for example, we know GeoServer implements the DescribeLayer
operation for WMS 1.1.1, but not for WMS 1.3.0, but another WMS
implementation may be doing the opposite... In a word, I'm just trying to
minimize the assumptions here.

On Mon, Mar 9, 2015 at 11:25 AM, mbarto <notifications@github.com> wrote:

> In mapcomposer/app/static/externals/gxp/src/script/plugins/WMSSource.js
> <https://github.com/geosolutions-it/MapStore/pull/586#discussion_r26025230>
> :
>
> > +    	var version;
> > +    	var baseParams = {
> > +    			SERVICE: "WMS",
> > +    			REQUEST: "DescribeLayer"
> > +    		};
> > +    	
> > +    	if (this.useCapabilities === true) {
> > +    		var describeLayerRequest = this.store.reader.raw.capability.request.describelayer;
> > +    		var wmsVersion = this.store.reader.raw.version;
> > +    		
> > +    		if (describeLayerRequest) {
> > +    			url = describeLayerRequest.href;
> > +    			version = wmsVersion;
> > +    		} else {
> > +    			url = this.url;
> > +    			version = this.switchWmsVersion(wmsVersion);
>
> I would make more explicit that in this case we are choosing the
> DEFAULT_WMS_VERSION and not switching to the other version.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geosolutions-it/MapStore/pull/586/files#r26025230>.
>